doc/doxygen/src/advanced-build-system-tricks.md

```diff
@@ -72,3 +72,17 @@ To do a repository wide analysis, you can use the script
 `dist/tools/buildsystem_sanity_check/save_all_dependencies_resolution_variables.sh`
 that will generate the output for all boards and applications.
 It currently take around 2 hours on an 8 cores machine with ssd.
+
+Generate Makefile.ci content                             {#generate-makefileci}
+============================
+
+Most applications and tests include a `Makefile.ci` to indicate which boards can
+not compile the application or test. The content for these files can be
+generated via the script in
+~~~~~~~~~~~~~~~~~~~
+make -C $APPLICATION_DIRECTORY generate-Makefile.ci
+~~~~~~~~~~~~~~~~~~~
+This will compile and link the application for every board available and record
+the result in the Makefile.ci. This requires the toolchain for every target to
+be available. The target supports using docker via the `BUILD_IN_DOCKER=1`
+variable.
```

Comment:
This is actually default behavior, if I recall correctly. This also makes sense: I (before updating newlib) often had smaller binaries than the CI toolchain, resulting in the output of `Makefile.ci` often not being useful. With `BUILD_IN_DOCKER=1` as default, the binary sizes should be the same as in the CI.

Reply:
Default it doesn't use docker to build for the `create-Makefile.ci` target here. I agree that it sounds like a sane default, but currently it is not the default.

Reply:
OK, sorry for the noise